lib/auth.coffee

```diff
@@ -29,6 +30,23 @@ getAuth = (deps, opts) ->
 	###
 	exports.twoFactor = twoFactor
 
+	userDetails = null
+	exports.getUserDetails = ->
+		return new Promise (resolve, reject) ->
```

Comment:
Most of the time, you don't really need to run `new Promise` yourself (with Bluebird especially), and there's a nicer answer, especially if you've already got a promise, like you do below.

In this case, you can simplify this to:

```coffeescript
Promise.try ->
  if userDetails
    return userDetails
  else
    return request.send
      ...
    .get('body')
    .then (body) ->
      userDetails = body
      return userDetails
```

`Promise.try` creates a promise, but acts like a normal `.then`. If you `return` then the promise resolves with that value. If you throw an exception, it rejects. If you return a promise, it uses the result of the promise, so you can skip the manual calls to `resolve`/`reject` entirely.